client/human/HumanClientApp.cpp

```diff
@@ -1479,3 +1480,75 @@ void HumanClientApp::OpenURL(const std::string& url) {
     // execute open command
     system(command.c_str());
 }
+
+void HumanClientApp::BrowsePath(const boost::filesystem::path& browse_path) {
+    if (browse_path.empty() || browse_path == "/") {
+        ErrorLogger() << "Invalid path: " << PathToString(browse_path);
+        return;
+    }
+
+    boost::filesystem::path full_path(browse_path);
+
+    try {
+        boost::filesystem::file_status status = boost::filesystem::status(full_path);
+        if (!boost::filesystem::exists(status)) {
+            std::string exists_debug_msg("Non-existant path: " + PathToString(full_path));
+            if (full_path.has_parent_path()) {
+                DebugLogger() << exists_debug_msg << ", trying parent directory";
+                BrowsePath(full_path.parent_path());
+            } else {
+                DebugLogger() << exists_debug_msg << ", aborting";
+            }
+            return;
+        }
+
+        // Validate as a canonical path
+        if (boost::filesystem::is_directory(status)) {
+            full_path = boost::filesystem::canonical(full_path);
+        } else {
+            // If given a file, use the files containing directory
+            DebugLogger() << "Non-directory target: " << PathToString(full_path) << ", using parent directory";
+            full_path = boost::filesystem::canonical(full_path.parent_path());
+        }
+
+        // Verify not a regular file
+        if (boost::filesystem::is_regular_file(full_path)) {
+            ErrorLogger() << "Target directory " << PathToString(full_path) << " is a regular file, given path argument: "
+                          << PathToString(browse_path);
+            return;
+        }
+
+    } catch (const boost::filesystem::filesystem_error& ec) {
+        ErrorLogger() << "Filesystem error when attempting to browse directory " << PathToString(full_path)
+                      << ": " << ec.what();
+        return;
+    }
+
+    if (full_path.empty()) {
+        ErrorLogger() << "Unable to determine directory for path " << PathToString(full_path);
+        return;
+    }
+
+    full_path.make_preferred();
+    std::string target(PathToString(full_path));
```

Comment:
This returns a UTF-8 string. For calling the Win32 API functions, you actually need a UTF-16 wstring path.

Reply:
Works for me:
````
    full_path.make_preferred();
    auto target = full_path.native();
    decltype(target) command;

    // Ampersand post-fixed to non-Windows platforms, prevents blocking until process exits
    // Double quotes around target to support paths containing spaces
    // Trailing slash post-fixed to prevent executing a file with same name(minus extension) as folder
#ifdef _WIN32
    command = L"start " + target;
#elif __APPLE__
    command = "open \"" + target + "/\" &";
#else
    command = "xdg-open \"" + target + "/\" &";
#endif

#ifdef _WIN32
    // https://msdn.microsoft.com/en-us/library/277bwbdz.aspx remarks all streams must be flushed or closed prior to std::system
    std::fflush(NULL);

    std::string utf8_command;
    utf8::utf16to8(command.begin(), command.end(), std::back_inserter(utf8_command));
    InfoLogger() << "Sending OS request to browse directory: " << utf8_command;

    int syscall_retval = _wsystem(command.c_str());

    DebugLogger() << "std::system(" << utf8_command << ") returned " << syscall_retval;

#else
    InfoLogger() << "Sending OS request to browse directory: " << command;
    int syscall_retval = std::system(command.c_str());
    DebugLogger() << "std::system(" << command << ") returned " << syscall_retval;
#endif
}
````

Reply:
Oh progress! I'd like to try codecvt over utf8 lib.
Conceptually, the path could be imbued and allow `path.string()` instead of `PathToString(path)`.  Since I'm not able to test that, I've only used it for the one log conversion.
Failing this one attempt, will use a more verbose form of your changes.